### PR TITLE
feat(frontend): add independent trace vs test data

### PR DESCRIPTION
**Pull request by jorgeepc**
This PR introduces the initial work to isolate the `trace tab data` vs the `test tab data`. The proposed solution was created with the following key points in mind:
- Follow recommended `redux architecture` to keep state minimal and derive additional data
- Use `memoized selectors` for optimization only when needed
- One single `derived data model` that could be used across UI components (not only DAG). Unify current types
- Take advantage of the "built-in" `node factory pattern` that is part of the DAG lib `react-flow`

### TestRun derived data

The idea of the custom `useSpanData` hook is to have one single component subscribed to updates in the redux state. This hook is responsible of providing span related data, such as: `span details` and span specific `analyzer errors`, `test specs` and `test outputs`. The selectors used by this hook are optimized and memoized only if needed. Here we are fixing critical performance issues that were introduced by our previous approach. 

```mermaid
classDiagram
    useSpanData <|-- TestRun_selectors
    TestRun_selectors <|-- TestRun_service
    useSpanData: span
    useSpanData: analyzerErrors
    useSpanData: testSpecs
    useSpanData: testOutputs
    class TestRun_selectors{
        Optimized Selectors with Memoization
        selectSpanById()
        selectAnalyzerErrorsBySpanId()
        selectTestSpecsBySpanId()
        selectTestOutputsBySpanId()
    }
    class TestRun_service{      
      Helper methods to transform test run data 
    }
```

#### DAG UI components (react-flow built-in factory)

```mermaid
classDiagram
    TraceSpanNode <|-- BaseSpanNode
    TraceSpanNode: useSpanData
    TraceSpanNode: -span
    TraceSpanNode: -analyzerErrors
    TestSpanNode <|-- BaseSpanNode
    TestSpanNode: useSpanData
    TestSpanNode: -span
    TestSpanNode: -testSpecs
    TestSpanNode: -testOutputs
    class BaseSpanNode{
        Base UI component
    }    
```

#### AttributeList UI components

```mermaid
classDiagram  
    class SpanDetail{
        useSpanData
        -span
        -analyzerErrors
        -testSpecs
        -testOutputs
    }
```

## Fixes

- #2780 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/src/services/TestRun.service.ts [hunk 1]
```diff
@@ -39,6 +42,59 @@ const TestRunService = () => ({
 
     return eventsWithoutPolling;
   },
+
+  getAnalyzerErrorsHashedBySpan(linterResult: LinterResult): TAnalyzerErrorsBySpan {
+    return linterResult.plugins
```
Comment (xoscar) [inline web/src/services/TestRun.service.ts:47]: What do you think about moving this logic to service functions?

File: web/src/components/Visualization/components/DAG/TestSpanNode/TestSpanNode.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import {INodeDataSpan} from 'types/DAG.types';
+import Footer from './Footer';
+import BaseSpanNode from '../BaseSpanNode/BaseSpanNode';
+
+interface IProps extends NodeProps<INodeDataSpan> {}
+
+const TestSpanNode = ({data, id, selected}: IProps) => {
+  const {span, testSpecs, testOutputs} = useSpanData(id);
+
+  const {matchedSpans} = useSpan();
```
Comment (xoscar) [inline web/src/components/Visualization/components/DAG/TestSpanNode/TestSpanNode.tsx:17]: Should we encapsulate this into `useShowAsCurrent` hook? 👀

**Review by xoscar — APPROVED** [review]
Awesome job @jorgeepc 🔥  I left some comments and concerns, let me know what you think

File: web/src/services/TestRun.service.ts (hunk 1 above)
Comment (jorgeepc) [reply web/src/services/TestRun.service.ts:47]: Hey @xoscar thanks for the review. What do you mean by service functions? This is already part of the TestRun service.

File: web/src/types/TestRun.types.ts [hunk 3]
```diff
@@ -6,3 +8,39 @@ export enum RunErrorTypes {
 }
 
 export type TTestRunState = NonNullable<TTestSchemas['TestRun']['state'] | 'WAITING' | 'SKIPPED' | 'FAILED'>;
+
+/* AnalyzerErrors by Span types */
+
+export type TAnalyzerError = {
+  ruleName: string;
```
[the remark this answers is not in the document]
Comment (jorgeepc) [reply web/src/types/TestRun.types.ts:15]: I think we can try it. Are you thinking about displaying some warning at the attribute level in the attribute list?

File: web/src/services/TestRun.service.ts (hunk 1 above)
Comment (xoscar) [reply web/src/services/TestRun.service.ts:47]: What haha I thought it was part of the selectors, sorry that was a mistake

File: web/src/types/TestRun.types.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (xoscar) [reply web/src/types/TestRun.types.ts:15]: Yeah, we can leave it for a future PR, I think we have a ticker for that

**Pull request merged**